File: WebAPI_ValeTransporte/API_VT/API_VT/Models/DespesaFuncionario.cs
using API_VT.Models.Entities;
using System.ComponentModel.DataAnnotations;

namespace API_VT.Models
{
    public class DespesaFuncionario
    {
        [Key]
        public int DespesaId { get; set; }
        public Despesa Despesa { get; set; }
        [Key]
        public int FuncionarioId { get; set; }
        public Funcionario Funcionario { get; set; }
        public int DiasTrabalhados { get; set; }
        public double CustoIndividual { get; set; }

        public DespesaFuncionario() { }

        public DespesaFuncionario(Despesa despesa, Funcionario funcionario, int diasTrabalhados, double custoIndividual)
        {
            Despesa = despesa;
            DespesaId = despesa.Id;
            Funcionario = funcionario;
            FuncionarioId = funcionario.Id;
            DiasTrabalhados = diasTrabalhados;
            CustoIndividual = custoIndividual;
        }
    }
}

File: ValeTransporte/src/app/models/DespesaFuncionario.ts
import { Despesa } from './Despesa';
import { Funcionario } from './Funcionario';

export class DespesaFuncionario {

    despesaId: number;
    despesa: Despesa;
    funcionarioId: number;
    funcionario: Funcionario;
    diasTrabalhados: number;
    custoIndividual: number;

    constructor() {
        this.despesaId = 0;
        this.funcionarioId = 0;
        this.diasTrabalhados = 0;
        this.custoIndividual = 0.0;
    }
}
